### PR TITLE
Fix #1949: Optimizes styling for onboarding final slide.

### LINKED ISSUE #1949: Optimise code in onboarding_slide_final.xml
This issue is tracking the optimisation work of `onboarding_slide_final.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all textviews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

3. If needed introduce common styles too in style.xml file.

While making the above changes the UI should not get affected and the end result should still be same as before (or an optimised version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by KMaragh**
## Explanation
Fixes #1949
Added common and specific styles to the appropriate styles.xml files, and updated the onboarding_slide_final.xml to use those new styles. Also added new dimensions to the appropriate dimens.xml files.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by KMaragh** [comment]
@rt4914 Recreated PR

File: app/src/main/res/layout-land/onboarding_slide_final.xml [hunk 1]
```diff
@@ -45,8 +45,7 @@
         android:fontFamily="sans-serif-medium"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide_final.xml:44]: In this view, you can remove `android:layout_height="wrap_content"` considering it is already defined in styles.xml
Check: `Heading1 -> TextView.Common1`

File: app/src/main/res/layout-land/onboarding_slide_final.xml [hunk 2]
```diff
@@ -59,11 +58,9 @@
         android:layout_marginStart="20dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide_final.xml:56]: In this view, you can remove `android:layout_height="wrap_content"` considering it is already defined in styles.xml
Check: `Subtitle1 -> TextView.Common1`

File: app/src/main/res/layout-land/onboarding_slide_final.xml [hunk 3]
```diff
@@ -76,13 +73,9 @@
...
         android:layout_marginTop="24dp"
         android:layout_marginEnd="52dp"
-        android:background="@drawable/primary_rounded_button"
-        android:fontFamily="sans-serif-medium"
-        android:minHeight="48dp"
         android:onClick="@{(v) -> viewModel.clickOnGetStarted()}"
         android:text="@string/get_started"
-        android:textColor="@color/white"
-        android:textSize="14sp"
+        style="@style/DefaultButton"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide_final.xml]: We don't need to create this new style as of now. You can revert this change.

File: app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml [hunk 4]
```diff
@@ -42,8 +42,7 @@
         android:fontFamily="sans-serif-medium"
         android:gravity="center_horizontal"
         android:text="@string/onboarding_slide_3_title"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml:43]: Apply all the changes to this file which I have mentioned previously.

File: app/src/main/res/layout-sw600dp-port/onboarding_slide_final.xml [hunk 5]
```diff
@@ -44,8 +44,7 @@
         android:gravity="center_horizontal"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-port/onboarding_slide_final.xml:43]: Apply all the changes to this file which I have mentioned previously.

File: app/src/main/res/layout/onboarding_slide_final.xml [hunk 6]
```diff
@@ -46,8 +46,7 @@
         android:gravity="center_horizontal"
```
Comment (rt4914) [inline app/src/main/res/layout/onboarding_slide_final.xml:45]: Apply all the changes to this file which I have mentioned previously.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 7]
```diff
@@ -6,4 +6,8 @@
   <dimen name="home_margin_min">64dp</dimen>
   <dimen name="recently_played_margin_max">96dp</dimen>
   <dimen name="recently_played_margin_min">64dp</dimen>
+
+  <dimen name="heading1_text_size">32sp</dimen>
+  <dimen name="subtitle1_text_size">24sp</dimen>
+  <dimen name="default_button_text_size">14sp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: This is not needed. Please revert the changes in this file.

File: app/src/main/res/values-sw600dp/dimens.xml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <dimen name="heading1_text_size">32sp</dimen>
+  <dimen name="subtitle1_text_size">24sp</dimen>
+  <dimen name="default_button_text_size">20sp</dimen>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp/dimens.xml]: This is not needed. Please revert the changes in this file.

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -62,10 +62,20 @@
...
   <dimen name="profile_view_not_added_margin">20dp</dimen>
   <dimen name="walkthrough_topic_summary_container_margin">20dp</dimen>
+  <dimen name="default_button_text_size">14sp</dimen>
+  <dimen name="heading1_text_size">24sp</dimen>
+  <dimen name="heading2_text_size">20sp</dimen>
+  <dimen name="heading3_text_size">18sp</dimen>
+  <dimen name="heading4_text_size">18sp</dimen>
+  <dimen name="heading5_text_size">14sp</dimen>
+  <dimen name="subtitle1_text_size">16sp</dimen>
+  <dimen name="subtitle2_text_size">14sp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: This is not needed. Please revert the changes in this file.

File: app/src/main/res/values/styles.xml [hunk 10]
```diff
@@ -238,11 +238,24 @@
...
   </style>
+
+  <style name="Button.Common" parent="Widget.AppCompat.Button">
+    <item name="android:layout_width">match_parent</item>
+    <item name="android:layout_height">wrap_content</item>
+    <item name="android:textColor">@color/white</item>
+    <item name="android:background">@drawable/primary_rounded_button</item>
+    <item name="android:fontFamily">sans-serif-medium</item>
+    <item name="android:minHeight">48dp</item>
+  </style>
```
Comment (rt4914) [inline app/src/main/res/values/styles.xml]: This is not needed. Please revert the changes in this file.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Left comments. Thanks @KMaragh

**Comment by rt4914** [comment]
@KMaragh have a look at the `4th` point in [this](https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change) and update the PR title accordingly

**Review by rt4914 — CHANGES_REQUESTED** [review]
Great. The code looks good.

There are couple of points:
1. Please follow the step mentioned in this comment to update the PR title https://github.com/oppia/oppia-android/pull/2014#issuecomment-711909654
2. Also, please do not `Resolve Comment`. The person who has started the conversation should resolve the comment. And you should always reply to comments. So please mark all comments as "unresolved" and reply that you have done that change. Then I will mark that as "Resolved".

Thanks. @KMaragh

File: app/src/main/res/layout-land/onboarding_slide_final.xml (hunk 1 above)
Comment (KMaragh) [reply app/src/main/res/layout-land/onboarding_slide_final.xml:44]: Made the change!

File: app/src/main/res/layout-land/onboarding_slide_final.xml (hunk 2 above)
Comment (KMaragh) [reply app/src/main/res/layout-land/onboarding_slide_final.xml:56]: Made the change!

File: app/src/main/res/layout-land/onboarding_slide_final.xml (hunk 3 above)
Comment (KMaragh) [reply app/src/main/res/layout-land/onboarding_slide_final.xml]: Made the change!

File: app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml (hunk 4 above)
Comment (KMaragh) [reply app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml:43]: Made the change!

File: app/src/main/res/layout-sw600dp-port/onboarding_slide_final.xml (hunk 5 above)
Comment (KMaragh) [reply app/src/main/res/layout-sw600dp-port/onboarding_slide_final.xml:43]: Made the change!

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 7 above)
Comment (KMaragh) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: Reverted

File: app/src/main/res/values-sw600dp/dimens.xml (hunk 8 above)
Comment (KMaragh) [reply app/src/main/res/values-sw600dp/dimens.xml]: Reverted

File: app/src/main/res/values/dimens.xml (hunk 9 above)
Comment (KMaragh) [reply app/src/main/res/values/dimens.xml]: Reverted

File: app/src/main/res/values/styles.xml (hunk 10 above)
Comment (KMaragh) [reply app/src/main/res/values/styles.xml]: Reverted

File: app/src/main/res/layout/onboarding_slide_final.xml (hunk 6 above)
Comment (KMaragh) [reply app/src/main/res/layout/onboarding_slide_final.xml:45]: Made the change!

**Comment by KMaragh** [comment]
Updated the title! I could have sworn I did it before 😅 , my mistake.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.
